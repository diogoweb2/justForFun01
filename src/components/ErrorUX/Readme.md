Button example:

```js
<div>
  <ErrorUX>Error Msg</ErrorUX>
</div>
```
