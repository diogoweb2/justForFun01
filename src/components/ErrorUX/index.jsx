import React from "react";
import PropTypes from "prop-types";

/**
 * Show a frindly generic erros to user
 */
const ErrorUX = ({ children }) => (
  <div
    className="slds-notify slds-notify_alert slds-theme_alert-texture slds-theme_error"
    role="alert"
  >
    <span className="slds-assistive-text">error</span>

    <h2>{children}</h2>
    <button
      className="slds-button slds-button_icon slds-notify__close slds-button_icon-inverse"
      title="Close"
    >
      <span className="slds-assistive-text">Close</span>
    </button>
  </div>
);

ErrorUX.propTypes = {
  children: PropTypes.node.isRequired
};
export default ErrorUX;
