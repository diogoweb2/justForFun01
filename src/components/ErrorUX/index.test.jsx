import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ErrorUX from "./index";

let wrapper;
describe("ErrorUX", () => {
  it("should render", () => {
    wrapper = shallow(<ErrorUX>test</ErrorUX>);
    expect(wrapper).toHaveLength(1);
  });

  it("should match snapshot...", () => {
    wrapper = shallow(<ErrorUX>OK</ErrorUX>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
