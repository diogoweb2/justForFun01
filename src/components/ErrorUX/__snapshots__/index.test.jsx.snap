// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorUX should match snapshot... 1`] = `
<div
  className="slds-notify slds-notify_alert slds-theme_alert-texture slds-theme_error"
  role="alert"
>
  <span
    className="slds-assistive-text"
  >
    error
  </span>
  <h2>
    OK
  </h2>
  <button
    className="slds-button slds-button_icon slds-notify__close slds-button_icon-inverse"
    title="Close"
  >
    <span
      className="slds-assistive-text"
    >
      Close
    </span>
  </button>
</div>
`;
